sway-core/src/abi_generation/fuel_abi.rs

```diff
@@ -51,7 +144,67 @@ impl TypeId {
             },
             engines,
             resolved_type_id,
+        )?;
+
+        let mut err: Option<ErrorEmitted> = None;
+
+        // Right now ABI renaming is only supported for enum and struct types.
+        let should_check_name = matches!(
+            *engines.te().get(resolved_type_id),
+            TypeInfo::Enum(_) | TypeInfo::Struct(_)
         );
+
+        if should_check_name {
+            let mut has_abi_name_attribute = false;
```

Comment:
Nitpick. Having a `mut` boolean here feels a bit over-complicated. One trivial possibility is to "calculate" `has_abi_name_attribute` after the `let (name, attribute_span)` or assign it within that same let: `let (has_abi_name, name, attr_span)`.